lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -432,6 +433,29 @@ newDBLayer logConfig trace mDatabaseFile = do
         , readPrivateKey = \(PrimaryKey wid) ->
               runQuery $ selectPrivateKey wid
 
+        {-----------------------------------------------------------------------
+                                       Pending Tx
+        -----------------------------------------------------------------------}
+
+        , removePendingTx = \(PrimaryKey wid) tid ->
+              ExceptT $ runQuery $
+              selectWallet wid >>= \case
+                  Just _ -> do
+                      metas <- selectPendingTxs wid (TxId tid)
+                      let checkStatus = not . any
+                              (\(TxMeta _ _ st _ _ _ _) -> st == W.Pending)
+                      case metas of
+                          [] ->
+                              pure $ Left $
+                              ErrRemovePendingTxNoSuchTransaction tid
+                          _ -> if (checkStatus metas) then
+                                  pure $ Left $
+                                  ErrRemovePendingTxTransactionNoMorePending tid
+                               else do
+                                  deletePendingTx wid (TxId tid)
+                                  pure $ Right ()
```

Comment:
Suggestion:

```hs
case metas of
    [tx] | isPending tx -> 
        deletePendingTx wid (TxId tid)
        pure $ Right ()

    [] ->
        pure $ Left $ ErrRemovePendingTxNoSuchTransaction tid

    otherwise -> 
        pure $ Left $ ErrRemovePendingTxTransactionNoMorePending tid
```

Reply:
done